File: scribble_kiddies/java_codes/Subsets.java
package java_codes;

public class Subsets
{
    public static void makeSubs(int n, int i,String ans)
    {
        if(i>n)
        {
            System.out.println(ans);
            return;
        }
        makeSubs(n,i+1,ans+Integer.toString(i));
        makeSubs(n,i+1,ans);
    }
    public static void main(String[] args)
    {
        makeSubs(3,1,"");
    }
     

}

File: scribble_kiddies/backTrack/subsets.py
class subsets:
    ans = []
    def generate(self,A,ANS,i,n):
        if i == n:
            self.ans.append(ANS)
            print(ANS)
            return
        ANS.append(A[i])
        self.generate(A,ANS,i+1,n)
        ANS.pop()
        self.generate(A,ANS,i+1,n)

A = subsets()
A.generate([1,2,3],[],0,3)
print(A.ans)
        